module6/ex28_tamperserve/ex28_tamperserve.py

```diff
@@ -0,0 +1,95 @@
+import hashlib
+import os
+import pickle
+import time
+
+from flask import Flask, request
+
+
+def get_sha1(pathfile):
+    with open(pathfile, 'rb') as f:
+        read_content = f.read()
+    file_hashed_sha1 = hashlib.sha1(read_content).hexdigest()
+    return file_hashed_sha1
+
+
+class FileInfo:
+    def __init__(self, filename, mtime, sha1):
+        self.filename = filename
+        self.mtime = mtime
+        self.sha1 = sha1
+
+    def __str__(self):
+        return f'FileInfo for {self.filename}, mtime {self.mtime}, sha1 {self.sha1}'
+
+    def __eq__(self, other):
+        return type(other) == type(self) and self.filename == other.filename
+
+
+class FileList:
+    def __init__(self, dir_path):
+        self.dir_path = dir_path
+        self.all_file_infos = []
+        self.last_timestamp = 0
```

Comment:
`scan` and `rescan` methods should be left in the class and maybe adapted a little.